Algorithm/QCAlgorithm.Python.cs

```diff
@@ -501,7 +501,7 @@ public FilteredIdentity FilteredIdentity(Symbol symbol, Resolution resolution, P
         /// <returns>A new FilteredIdentity indicator for the specified symbol and selector</returns>
         public FilteredIdentity FilteredIdentity(Symbol symbol, TimeSpan resolution, PyObject selector = null, PyObject filter = null, string fieldName = null)
         {
-            var name = string.Format("{0}({1}_{2})", symbol, fieldName ?? "close", resolution);
+            var name = $"{symbol}({fieldName ?? "close"}_{resolution.ToStringInvariant(null)})";
```

Comment:
The `IFormattable.ToStringInvariant(string format)` extension method is what is being invoked here since `TimeSpan` does not implement `IConvertible` -- so we're essentially calling `IFormattable.ToString(format: null, formatProvider: CultureInfo.InvariantCulture)`. We can't have a default value for `format = null` in the extension method since it would cause an ambiguity, also, specifying a null `format` should be have the same as _not_ specifying a format, so this maintains expected invariants.

Reply:
So the reason this pattern was chosen was due to .net 4.5.2 not having `FormattableString`, which is what string interpolation (`$"my-{interpolated}-string"`) compiles into. Sadly, in .net 4.5.2's FxCop won't create warnings/errors for `IFormattable`/`CultureInfo` not being provided in interpolated strings, so we'll need to be vigilant in our code reviews. In an effort to combat this, I found a nuget package that adds `FormattableString` to enable the pattern preferred by the dotnet core/rosyln teams for handling invariant cultured strings:
``` csharp
using static System.FormattableString

// ... somewhere inside a function
var myCulturedString = Invariant($"my-{interpolated}-string")
```
and this will perform the formatting of each argument using `CultureInfo.InvariantCulture`.

In an effort to keep all of our formatting going through a common code path, I've added `StringExtensions.Invariant` which currently simply pipes directly through to `FormattableString.Invariant`, so it enables the exact same syntax, but felt it beneficial to have all of the cases go through a single LEAN method, so if we need to change/modify the behavior, it should be as easy as updating this single method definition.

Since changing the pattern to support the `using static` (can't be extension method), many of these cases with a `null` format string have been removed as those cases are no longer using the `StringExtensions.ToStringInvariant(string)` method, but instead using `FormattableString.Invariant(FormattableString)`